cgi-bin/DW/Controller/Admin/SendMail.pm

```diff
@@ -171,5 +171,122 @@ sub index_controller {
     return DW::Template->render_template( 'admin/sendmail/index.tt', $rv );
 }
 
+sub lookup_controller {
+    my ( $ok, $rv ) = controller( privcheck => \@privs, form_auth => 1 );
+    return $rv unless $ok;
+    my $remote = $rv->{remote};
+
+    my $scope = sub { return '/admin/sendmail/lookup.tt' . $_[0] };
+
+    my $enhance_row = sub {
+        my ( $row ) = @_;
+        # link to view full message
+        $row->{msgurl} = '?id=' . $row->{msgid};
```

Comment:
Hrrm, are we always sure that the current page has no arguments? (I think if there are, this will lead to an invalid url)

I suggest using LJ::create_url( undef, ...) instead to build the url.